src/cli/command-processor.ts

```diff
@@ -0,0 +1,225 @@
+
+  /**
+   * @param requestId  The request id for a triggered Execute request.
+   * @param localDeviceId  The localDeviceId for a triggered Execute request.
+   * @param executeCommand  The single command string to set in the triggered Execute request.
+   * @param params  The params array for the single Execute command.
+   * @param customData  The customData array for the single Execute command.
+   * @returns  A new ExecuteCommand instance.
+   */
+  constructor(
+    requestId: string,
+    localDeviceId: string,
+    executeCommand: string,
+    params: Record<string, unknown>,
+    customData: Record<string, unknown>
+  ) {
+    this.requestId = requestId;
+    this.localDeviceId = localDeviceId;
+    this.executeCommand = executeCommand;
+    this.params = params;
+    this.customData = customData;
+  }
+}
+
+export class CommandProcessor {
+  private worker: Worker;
+  constructor(worker: Worker) {
+    this.worker = worker;
+  }
+
+  // Wrapper for argument parsing with yargs.
+  private async parseIntent(
+    userCommand: string
+  ): Promise<IntentMessage | void> {
+    const argv = await yargs()
+      .command('identify', 'Trigger an Identify command.', yargs => {
+        return yargs
+          .option('request_id', {
+            describe: 'The request Id',
+            type: 'string',
+            demandOption: true,
+          })
+          .option('discovery_buffer', {
+            describe: 'The IDENTIFY dicovery buffer represented as a string',
+            type: 'string',
+            demandOption: false,
+          })
+          .option('device_id', {
+            describe: 'The device Id',
+            type: 'string',
+            demandOption: false,
+          });
+      })
+      .command('execute', 'Trigger an Execute command.', yargs => {
+        return yargs
+          .option('local_device_id', {
+            describe: 'The local device Id',
+            type: 'string',
+            demandOption: false,
+          })
+          .option('command', {
+            describe: 'The execute command to send to device_id',
+            type: 'string',
+            demandOption: false,
+          })
+          .option('params', {
+            describe:
+              'The params argument for the execute command, in JSON format',
+            type: 'string',
+            default: '{}',
+            demandOption: false,
+          })
+          .option('custom_data', {
+            describe:
+              'The customData argument for the execute command, in JSON format',
+            type: 'string',
+            default: '{}',
+            demandOption: false,
+          });
+      })
+      .command('exit', 'Exit the command line interface.')
+      .parse(userCommand, {}, (error, argv) => {
+        if (error !== null) {
+          throw error;
+        }
+        return argv;
+      });
+
+    /**
+     * Validate arguments and return an IntentMessage
+     */
+    switch (argv._[0]) {
```

Comment:
Is `argv._` the right way to get the list of arguments?

Reply:
I believe so - there is usage in the yargs documentation.